Source/JavaScriptCore/runtime/WaiterListManager.cpp

```diff
@@ -316,7 +315,7 @@ void WaiterListManager::unregister(JSGlobalObject* globalObject)
         list->removeIf(listLocker, [&](Waiter* waiter) {
             if (waiter->isAsync()) {
                 auto ticket = waiter->ticket(listLocker);
-                if (ticket->globalObject == globalObject) {
+                if (ticket->globalObject() == globalObject) {
```

Comment:
This does not work when the target is also dead at this point.

Reply:
Yes, encountered that crash. Fixed.